File: web/src/types/lead.ts

/**
 * Lead domain types — mirror the Go backend domain model.
 * Field names use camelCase (frontend convention), mapping to the
 * snake_case json tags on the Go domain.Lead struct.
 */

export type LeadStatus =
  | 'new'
  | 'assigned'
  | 'in_progress'
  | 'visited'
  | 'closed_won'
  | 'closed_lost'

export type CustomerType =
  | 'retail'
  | 'wholesale'
  | 'hospitality'
  | 'institutional'
  | 'other'

export interface Lead {
  id: string
  title: string
  description: string
  status: LeadStatus
  assigneeId: string
  teamId: string
  customerId: string
  customerType: CustomerType
  company: string
  industry: string
  location: string
  valueCents: number
  initials: string
  createdAt: string
  updatedAt: string
  deletedAt?: string | null
}

export interface CreateLeadInput {
  title: string
  description?: string
  status?: LeadStatus
  assigneeId: string
  teamId: string
  customerId: string
  customerType: CustomerType
}

export interface UpdateLeadInput {
  id: string
  title?: string
  description?: string
  status?: LeadStatus
  assigneeId?: string
  teamId?: string
  customerId?: string
  customerType?: CustomerType
}

export interface LeadListParams {
  page?: number
  limit?: number
  status?: LeadStatus
  assigneeId?: string
}

File: internal/domain/lead.go
package domain

import "time"

// Lead represents a sales opportunity that a rep visits at a customer location.
type Lead struct {
	ID           string       `json:"id"`
	Title        string       `json:"title"`
	Description  string       `json:"description"`
	Status       LeadStatus   `json:"status"`
	AssigneeID   string       `json:"assigneeId"` // User.ID of the assigned rep
	TeamID       string       `json:"teamId"`     // Team.ID that owns this lead
	CustomerID   string       `json:"customerId"` // Customer.ID this lead is associated with
	CustomerType CustomerType `json:"customerType"`
	// Company is the name of the company this lead is associated with.
	Company string `json:"company"`
	// Industry is the industry sector of the lead's company.
	Industry string `json:"industry"`
	// Location is the geographic location of the lead.
	Location string `json:"location"`
	// ValueCents is the estimated deal value in cents (use cents to avoid float precision issues).
	ValueCents int64 `json:"valueCents"`
	// Initials are the display initials for the lead contact (e.g. "JD" for John Doe).
	Initials  string    `json:"initials"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
	// DeletedAt is set when the lead is soft-deleted. Nil means active.
	DeletedAt *time.Time `json:"deletedAt,omitempty"`
}
